### ISSUE TITLE
Star Soldier Enhanced disappearing sprite. Transparancy bit not working?

### ISSUE THREAD

**Issue by TFHFony** [ppeccin/WebMSX#98]

**Comment by NullPointerReference** [comment]
It does not seem to happen with the non-OPLL versions of the smooth-scrolls; in that case the sprite correctly turns black; so I kinda assume this has to do with the modification (it's a different ROM-type; so could be a mapping/addressing issue perhaps ?) and not a WebMSX bug.

See for example: https://download.file-hunter.com/Games/MSX2/ROM/Star%20Soldier%20-%20Hudson%20Soft%20(1986)%20[Smooth%20Scroll].zip

**Comment by ppeccin** [comment]
I see...

Can you point us to the "faulty" version of the ROM?
Also, do you know what would be the correct type for it?

Thanks

**Comment by NullPointerReference** [comment]
Yes; the Faulty ROM is: https://download.file-hunter.com/Games/MSX2/ROM/Star%20Soldier%20-%20Hudson%20Soft%20(1986)%20[Smooth%20Scroll%20&%20OPLL%20v0.2].zip

And the ROM type is: KonamiSCC.

**Comment by ppeccin** [comment]
Can you describe a little more what the faulty behavior is?
I played both ROMs, and at some moment the player ship disappears in both
versions. It looks like both are faulty.

Are you sure the problem does not happen in one of the versions?

Paulo

**Comment by NullPointerReference** [comment]
Hi Paolo, yes, if I run the 'smooth scrolling non-OPLL' version (from the link I provided) through this link: https://download.file-hunter.com/assets/webmsx.html?url=https%3A%2F%2Fdownload.file-hunter.com%2FGames%2FMSX2%2FROM%2FStar%2520Soldier%2520-%2520Hudson%2520Soft%2520(1986)%2520%5BSmooth%2520Scroll%5D.zip 

I.e. go to https://download.file-hunter.com, search for 'Star Soldier' and click the RUN button on the results - the latter shows a black-sprite (and bullets as normal)  whilst the 'opll v0.x' versions both don't show any ship-sprite NOR bullet-sprites (I didn't mentioned the latter before) at all at those moments the ship is supposed to turn black.  Please do note that the 'RUN' button sets the 'unlimited sprites' option by default; but turning that off doesn't affect the behavioural difference.

Also note that the 'opll v0.x' version need to have their ROM type set to 'KonamiSCC' manually when clicking 'RUN'.

**Comment by ppeccin** [comment]
I see...
I didn't understand at first that the ship becoming black was expected. I
looked like a bug!

Peccin

**Comment by NullPointerReference** [comment]
No, the ship turning black is in the original as well - and also happens in i.e. OpenMSX.  I did try the 'smooth scrolling with opll v0.2' with OpenMSX (CBIOS, 2+) btw, and it DID show the black ship there with that same ROM too.  Then I retested it another time with WebMSX (2+ too, but it doesn't seem to matter what machine you pick) and the ship does still 'disappear' on WebMSX, so there is indeed some difference in emulation.   Could it be related to https://github.com/ppeccin/WebMSX/issues/81 perhaps ? Or something to do with the cartridge format after all ?

PS: Sorry for the 'noise' I really thought it was a bad ROM-mod initially, given the non-OPLL versions just work fine on WebMSX.
